src/csharp/Microsoft.Spark/Utils/DependencyProviderUtils.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+
+namespace Microsoft.Spark.Utils
+{
+    internal class DependencyProviderUtils
+    {
+        internal static readonly string s_filePattern = "dependencyProviderMetadata_*";
+
+        internal static void Serialize(Metadata metadata, string path)
```

Comment:
I would expose this as a member function of `Metadata`.

Reply:
Moved.